### ISSUE TITLE
ENH: Share some image driver routines with other drivers

### ISSUE THREAD

**Issue by jhamman** [UW-Hydro/VIC#369]
VIC's driver structure looks something like this:

```
| - drivers
| --- shared 
| --- classic
| --- python
| --- image
```

I'm proposing a minor reorganization to cut down on duplicate code as we begin to add other drivers that will use significant portions of the Image Driver.

```
| - drivers
| --- shared_all
| --- shared_image
| --- classic
| --- python
| --- image
| --- cesm
| --- wrf_hydro
| --- nasa_lis
```

- `vic/drivers/shared` becomes `vic/drivers/shared_all` and includes code shared between all drivers.
- code in `vic/drivers/image` that is general and will be used by any driver running VIC in some version of the Image Driver goes in `vic/drivers/shared_image`. Most notably, this will include the netCDF routines for reading and writing history and statefiles.  Drivers (e.g. `nasa_lis`) do not have to use this code but it will be available to them.  The reason not to put it all in the existing `shared` directory is some of this code has dependencies not required by the Classic Driver.

This will significantly cut down on the code in the drivers that look mostly like the Image Driver but differ in coupling (e.g. #265).

**Comment by tbohn** [comment]
The names `image` and `shared_image` are somewhat similar and I find it potentially confusing - "image" appears to be both a general category (referring to image *mode*) and a specific flavor of that category.  Perhaps "image" should be changed to something like "image_default" or some other name that indicates that it is one of many image mode drivers?

**Comment by tbohn** [comment]
(otherwise I think your new structure is a good idea)

**Comment by bartnijssen** [comment]
That's a good idea. Maybe something that conveys that the `image` mode is a standalone driver (all the other `image` modes are part of a larger framework).

`vic_image_standalone` is a bit off a mouthful (as opposed to `vic_classic` or `vic_image`. I suggest moving ahead with the directory split (we can always rename that), but to keep @tbohn's comment in mind. If anyone has a good title that'd be great.

**Comment by tbohn** [comment]
How about:
vic_default
vic_generic
vic_vanilla
vic_nilla

**Comment by jhamman** [comment]
@tbohn - do you have any branches in progress right now (e.g. statefile stuff from #367)?  If you haven't started that yet, I'll bust this issue out real quick.

**Comment by jhamman** [comment]
Also - I'm not sure I can bring myself to calling anything `vic_nilla`.  Way to gangsta, VIC is more of a 1990s minivan.

**Comment by tbohn** [comment]
I haven't yet started the statefile mods to VIC code - focusing on tonic first.  So bust a move.

How about "vic_standalone" without the word "image" in it, since it's already going to be under the "image" umbrella?

**Comment by jhamman** [comment]
closed via #392
